File: 1528-shuffle-string/1528-shuffle-string.c
char * restoreString(char * s, int* indices, int indicesSize){

    
    // arr(indices[i]) = s[i];
    

    char *arr = (char*)malloc(sizeof(char)*(indicesSize+1));
    for (int i = 0; i < indicesSize; i++){
        int t = indices[i];
        arr[t] = s[i]; 
        }   
    arr[indicesSize] = '\0'; 
    return arr;
}

File: 1528-shuffle-string/1528-shuffle-string.java
class Solution {
    public String restoreString(String s, int[] indices) {
        char[] arr = new char[s.length()];
        for (int i = 0; i < arr.length;i++){
            arr[indices[i]] = s.charAt(i);
        }
        return String.valueOf(arr);

    }
}